### PR TITLE
Chore(Update) planet attributes for tourism + humans

### PR DESCRIPTION
**Pull request by MasterOfGrey**
## Summary

I wanted to have a branch in a conversation for a new mission based on whether you were on a fully Hai world or one with a fairly consistent human presence.

In doing so I found that a) there was no attribute for identifying that, and b) there were a bunch of attributes missing from worlds they should've been on.

So this just adds some attributes that need to be there.

### CODE REVIEW

**Review by petervdmeer — APPROVED** [review]

**Review by warp-core — COMMENTED** [review]
What does `hai-human presence` mean and why is there a hyphen in it?

**Comment by MasterOfGrey** [comment]
It is for a Hai world with human presence (to be distinct from Human worlds with Hai presence, should that ever come to pass), and it has a Hyphen because I originally put a colon and the style checker blew up at me.

File: data/map planets.txt [hunk 1]
```diff
@@ -3036,11 +3036,11 @@ planet "New Washington"
...
-	attributes hai textiles urban
+	attributes hai textiles urban "hai-human presence"
 	landscape land/myrabella7
 	description `Newhome's gravity, climate, and geography is very similar to the Hai homeworld. Several billion Hai live here. Most dwell in large factory cities, but there are also ranches in the undeveloped areas where the Hai raise animals that are larger than elephants but produce wool similar to sheep. Shearing them is a dangerous undertaking, but their wool can be used to make excellent, long-lasting clothing.`
 	spaceport `The roads in the spaceport are very wide, so that Hai ranchers can lead their massive, sheep-like beasts along them. The animals seem docile enough, but their size is alarming.`
-	spaceport `	Most of the shops only sell clothing made for the Hai, but a few enterprising artisans have begun making clothing for humans as well: larger dimensions, and no tail holes in the pants.`
+	spaceport `	Most of the shops only sell clothing made for the Hai but, due to the growing human population in the spaceport district, a few enterprising artisans have begun making clothing for humans as well: larger dimensions, and no tail holes in the pants.`
```
Comment (EjoThims) [inline data/map planets.txt:3043]:
```suggestion
	spaceport `	Most of the shops only sell clothing made for the Hai, but, due to the growing human population in the spaceport district, a few enterprising artisans have begun making clothing for humans as well: larger dimensions, and no tail holes in the pants.`
```

Still need that first comma before the conjunction.

Comment (EjoThims) [inline data/map planets.txt:3043]:
```suggestion
	spaceport `	Most of the shops only sell clothing made for the Hai, but a few enterprising artisans have responded to the growing human population in the spaceport district and begun making clothing for humans as well: larger dimensions, and no tail holes in the pants.`

We can skip that comma mess, though, with some reorganizing, but it does change the emphasis of the framing.

**Review by EjoThims — CHANGES_REQUESTED** [review]
Makes sense, just a minor error in the expanded text.

File: data/map planets.txt (hunk 1 above)
Comment (warp-core) [reply data/map planets.txt:3043]: The suggestion isn't closed properly.

Comment (MasterOfGrey) [reply data/map planets.txt:3043]: It shouldn't though, because the added section is now a clause?

Comment (MasterOfGrey) [reply data/map planets.txt:3043]:
```suggestion
	spaceport `	Most of the shops only sell clothing made for the Hai, but a few enterprising artisans have responded to the growing human population in the spaceport district and begun making clothing for humans as well: larger dimensions, and no tail holes in the pants.`
```

Just copy-pasting that suggestion for proper closure - I much prefer this one. My original change was simply a least-impact approach given it was just a smidge of additional clarity to support the attribute tag being added.

Comment (EjoThims) [reply data/map planets.txt:3043]: Whoops.

**Comment by warp-core** [comment]
> It is for a Hai world with significant human presence (to be distinct from Human worlds with Hai presence, should that ever come to pass), and it has a Hyphen because I originally put a colon and the style checker blew up at me.

Why not just `human presence` or `human population` for the attribute? The fact that these are Hai worlds should be evidenced by the presence of the `hai` attribute.
A hyphen really isn't a good substitute for a colon, `hai-human presence` makes it sound like this is some sort of joint enterprise where both sides have gone in together on a colony; you might say Sabira Eseskrai is a `korath-wanderer` world.

**Comment by W1zrad** [comment]
@MasterOfGrey Wrap brings up a good point. Want to make the change, or do you have any specific opposition?

**Comment by MasterOfGrey** [comment]
My concern is about distinctiveness. `human presence` could be anywhere with humans, and what if we end up with humans sharing a world with another species down the line? Then you'd have to separate them.

I'm concerned about setting a precedent of generic attributes for specific things, when a specific attribute is perfectly equally viable.

If the concern is that "A hyphen really isn't a good substitute for a colon" - then perhaps someone could identify how to use a colon without tripping up the style checker?

**Review by EjoThims — APPROVED** [review]

**Comment by warp-core** [comment]
Then do something like `human presence on hai world` (though, the meaning you want to convey with this attribute currently makes it entirely redundant with the `hai` attribute which already indicates that this is a hai world, which is why I don't think it is actually necessary to be so specific.)

**Comment by MasterOfGrey** [comment]
> Attributes should be used similarly to tags, as in multiple basic tags, which can be combined in various ways to create more advanced checks. Rather than "hai + human" as one tag, you have a tag for "hai" and a tag for "human" and can check for one, or the other, or both, as necessary.

I mean, the intrinsic problem with this approach is that if you have things looking for "human" only, then they can trigger anywhere that has human - even if that attribute is only applied to a location for the purpose of having "hai + human", necessitating further source filter components to further restrict it. Which is counter-productive.

> Then do something like `human presence on hai world` (though, the meaning you want to convey with this attribute currently makes it entirely redundant with the `hai` attribute which already indicates that this is a hai world, which is why I don't think it is actually necessary to be so specific.)

> So as warp suggested above, any non-human world with a human presence would have a "human presence" tag, allowing anyone to check for, say, "hai" _and_ "human presence," or "hai" and _exclude_ "human presence," or any other combination as necessary, without needing a specific tag for each possible combination.

And sure, if you have something that *isn't* used anywhere else, as "human presence" obviously is not currently used somewhere else, then that does work but:

- You're still adding a new attribute tag either way. So choosing one option over the other is entirely cost-neutral.
- Needing to use 2 attributes _doubles_ the complexity and effort checked on every single filter used everywhere else in the game.

These together mean that using a specific attribute is more efficient.

_IF_ it gets used later on other alien worlds, then once "Human Presence" existed on >2 alien species worlds, then there would be marginal efficiency gain from using combination of attributes - but that runs afoul of another issue which is:

- People constantly insist that filters be shortened to be as simple as possible - so if "Human Presence" only appears on Hai worlds, someone is gonna end up having all those filters _only_ refer to "Human Presence" and not "Hai + Human Presence"
- And THEN, if it gets used somewhere else it would then also necessitate going back and updating all those filters again, which is a great way to introducing bugs and unintended behaviours.

So *the most efficient* usage here is to use a specific attribute, and the only possible defence for future-proofing it to (hopefully) be more efficient in a hypothetical future, is to explicitly change the approach that is applied constantly to simplifying filter criteria.

Ergo:
Efficient Solution: specific attribute
Alternative Solution: mandate change to development practice

**Comment by warp-core** [comment]
Looking at the output of the style checker, its complaints were not about the presence of colons, but instead the lack of space following the colon.
`hai:human presence` should have been `hai: human presence`.
That passes the style check (https://github.com/warp-core/endless-sky/pull/20) and doesn't have the issue I mentioned earlier where it suggests an entirely different meaning (one of some sort of jointly developed colony a-la Sabira Eseskrai).

**Comment by W1zrad** [comment]
> IF it gets used later on other alien worlds, then once "Human Presence" existed on >2 alien species worlds, then there would be marginal efficiency gain from using combination of attributes

Using this would be ideal (Hecter's "tag" explanation seems the most logical way to go about this, especially because that reduces the amount of specific tags we have), but I don't think this is feasible given we still have a ways to go on development. Having to go back and update missions based on future planets that get tagged sounds... not fun. Obviously we'd prefer that we catch these possibilities in the PR phase, but it seems more likely that we'll miss a few and then end up having to bugfix.

I'm good with merging this as-is then. Will give it a little time if people want to continue deliberating, but if I don't hear anything in the next few days I'll merge.

File: data/map planets.txt [hunk 2]
```diff
@@ -120,7 +120,7 @@ planet Alix
 	description `Alix is a small moon of the gas giant Pathera, lacking a breathable atmosphere but rich in minerals. While settlements on the moon have been destroyed to the extent that all that remains are debris scattered across the terrain, some automated facilities are still functioning and see a constant flow of cargo drones that stop by the moon to gather raw materials before quickly bringing them to the nearby station's reserves.`
 
 planet Allhome
-	attributes farming hai "human tourism" urban
+	attributes farming hai "human tourism" "hai: human presence" urban
```
Comment (warp-core) [inline data/map planets.txt:123]:
```suggestion
	attributes farming hai "hai: human presence" "human tourism" urban
```

File: data/map planets.txt [hunk 3]
```diff
@@ -1786,7 +1786,7 @@ wormhole "Graveyard Wormhole (Unstable)"
 	color "wormholes: Ember Waste"
 
 planet Greenbloom
-	attributes hai "hai tourism" "human tourism"
+	attributes hai "hai tourism" "human tourism" "hai: human presence"
```
Comment (warp-core) [inline data/map planets.txt:1789]:
```suggestion
	attributes hai "hai tourism" "hai: human presence" "human tourism"
```

File: data/map planets.txt [hunk 4]
```diff
@@ -1816,7 +1816,7 @@ planet Greenrock
 		fleet "Large Northern Pirates" 13
 
 planet Greenview
-	attributes hai station
+	attributes hai station "hai tourism" "human tourism" "hai: human presence"
```
Comment (warp-core) [inline data/map planets.txt:1819]:
```suggestion
	attributes hai station "hai tourism" "hai: human presence" "human tourism"
```

File: data/map planets.txt [hunk 5]
```diff
@@ -1825,7 +1825,7 @@ planet Greenview
 	security 0.5
 
 planet Greenwater
-	attributes farming fishing hai "human tourism"
+	attributes farming fishing hai "human tourism" "hai: human presence"
```
Comment (warp-core) [inline data/map planets.txt:1828]:
```suggestion
	attributes farming fishing hai "hai: human presence" "human tourism"
```

File: data/map planets.txt [hunk 6]
```diff
@@ -1844,7 +1844,7 @@ planet "Gresku Fodar"
 	security 0
 
 planet Greymoon
-	attributes hai station
+	attributes hai station "human tourism"
```
Comment (warp-core) [inline data/map planets.txt:1847]:
```suggestion
	attributes hai "human tourism" station
```

File: data/map planets.txt [hunk 7]
```diff
@@ -3036,11 +3036,11 @@ planet "New Washington"
 		fleet "Large Militia" 3
 
 planet Newhome
-	attributes hai textiles urban
+	attributes hai textiles urban "hai: human presence"
```
Comment (warp-core) [inline data/map planets.txt:3039]:
```suggestion
	attributes hai "hai: human presence" textiles urban
```

File: data/map planets.txt [hunk 8]
```diff
@@ -4184,7 +4184,7 @@ planet Sunracer
 		fleet "Large Syndicate" 10
 
 planet Swiftsong
-	attributes hai "hai tourism" station
+	attributes hai "hai tourism" "human tourism" "hai: human presence" station
```
Comment (warp-core) [inline data/map planets.txt:4187]:
```suggestion
	attributes hai "hai tourism" "hai: human presence" "human tourism" station
```

**Review by warp-core — COMMENTED** [review]
Alphabetical order

**Pull request merged**